### ISSUE TITLE
Hyper lowers a different mixer than hyper-connections: layer norm, 1+linear gates, and a lane sum instead of per-lane RMS, sigmoid low-rank gates, mean collapse, and 2*sigmoid injection

### ISSUE THREAD

**Issue by nm-z** [nm-z/recipe-dev#673]
### Intended scope

`hyper(lanes, rank, &model)` (#569, #598) exists so a Recipe model can express hyper-connections: a stream of `lanes` copies of the width, a data-dependent read gate that mixes the lanes into one branch input, a per-lane write gate that scatters the branch output back, and the same read at the head. A real checkpoint that declares this structure exists: its GGUF metadata carries `<arch>.hyper_connection.count` = 4 and `<arch>.hyper_connection.low_rank` = 320, which the GGUF metadata reader parses into `hyper_lanes`/`hyper_rank` at recipe.rs:5113-5114. The reference mixer for that checkpoint, the `build_hc_mix` and `build_hc_combine` graph functions in its architecture's model file under archy:/home/nate/llama.cpp-master/src/models/, is:

```
xn     = rms_norm_per_lane(x, eps) * gamma[lanes*width]      // statistics over one lane, one gamma over the whole stream
gate   = sigmoid(W_up · silu(W_down · xn / lanes))           // W_down [stream, rank], W_up [rank, stream], no bias
mixed  = mean over lanes of (xn * gate)                       // width wide
inject = 2 * sigmoid(W_inject · xn / lanes)                   // [lanes] per token, no bias
x[lane] += block_out * inject[lane]
```

The head mixer is the same without `inject`. A complete hyper block must compute exactly this so a hyper-connected checkpoint binds and reproduces its forward.

### Actual scope

`lower_gates` (recipe.rs:9908-9930, identical in PR #598 and after commit 3ae5a81b, "Allow Recipe submodels inside hyper-connections") lowers:

- `Normalize` with `arguments(1.0, epsilon)` (recipe.rs:9914). Mode 1.0 is `BlockNormalization::Layer` (recipe.rs:6872): mean subtraction plus variance. Width and span arguments are zero, which `normalize_width`/`normalize_span` (recipe.rs:4029-4037) resolve to the whole row, so statistics are taken over all `lanes*width` channels, not per lane. The node is pushed with 0 parameters: there is no gamma.
- `lower_project(graph, true, rank)` then `lower_project(graph, true, channels)` (recipe.rs:9917-9918). `true` adds a bias (recipe.rs:9646-9651). No `silu`, no `1/lanes` scale, no `sigmoid`.
- `lower_offset_one` (recipe.rs:9931-9936): gate = 1 + W·(...). The gate is unbounded and affine.
- `Read` consumes the raw stream `stream`, not the normalized `xn` (recipe.rs:9884-9885). `read_forward_body` (amd-nv-cpu.ll:990-999) accumulates `gate*value` over lanes and stores the sum; there is no division by `lanes`.
- Write gate: `lower_project(graph, true, lanes)` plus `lower_offset_one` (recipe.rs:9925-9926): 1 + W·xn + b, not `2*sigmoid(W·xn/lanes)`.
- `lower_collapse` (recipe.rs:9938-9945) reuses the same gates at the head.

The metadata fields `hyper_lanes`/`hyper_rank` are parsed and never consumed by any lowering (rg over recipe.rs finds only the struct and parser).

### Concrete consequence

A real hyper-connected GGUF checkpoint (4 lanes, rank 320, width 2560, 48 layers) cannot be bound to a `hyper(4, 320, ...)` block and, if forced, computes the wrong forward:

- Parameter shapes do not match. GGUF `blk.N.hc_attn_norm {10240}` has no slot (Recipe's Normalize carries 0 weights). `hc_attn_down {10240,320}` and `hc_attn_up {320,10240}` have no biases; Recipe's projections carry 320 + 10240 extra bias parameters. `hc_attn_inject {10240,4}` has no bias; Recipe's write projection has 4. A whole-model binder that checks `node.weights()` against the tensor element count fails on every hyper block and at the head.
- Even with weights forced in, the output is wrong at every layer: layer norm subtracts a row mean over 10240 channels that RMS does not; `1 + W·x` is not `sigmoid(...)`; the read sums 4 lanes where the reference takes their mean (a 4x scale on the branch input at every one of 48 layers); the injection is `1 + W·xn` where the reference is `2*sigmoid(W·xn/4)`, which is bounded in (0, 2). Logits diverge from the first layer, so any inference or evaluation of that checkpoint through Recipe is incorrect, not merely different.
- For training from scratch, the affine gates are unbounded, so a lane can scale without limit. No measurement of this exists; this is an inference from the form of the gate.

### Why it is partial

`hyper` was written on `minimal` at 10148e25, where the only per-row normalization was `Layer` and the only elementwise ops used by the residual path were `Add`. RMS mode landed separately (#588, 9e608639) and was never folded back in. The acceptance boundary on #569 and in the #598 thread was (a) `hyper(1, 0, ...)` bit identical to `res(...)` and (b) CPU equal to nv0. An affine `1 + W·x` gate with `rank` 0 satisfies both trivially because `rank` 0 pushes no node at all (recipe.rs:9909-9911). Nothing in the boundary exercised a nonzero rank against a reference, so the functional form of the gate, the collapse rule and the injection scaling were never checked. The PR body and README describe the design in words that fit both mixers ("gates come from a rank bottleneck on the normalized stream"), which hid the difference.

### Required correction

One generalized `hyper(lanes, rank, &model)`; no architecture-specific path and no second API.

1. Normalize: `Primitive::Normalize` in `Rms` mode with `width = channels / lanes`, `span = channels`, and `parameters = channels` (per-lane statistics through the existing group emitter, one gamma over the whole stream, trainable). Epsilon from the model, not the build-time constant (see the model-owned-epsilon design note).
2. Read gate: `lower_project(graph, false, rank)`, scale by `1/lanes`, `silu`, `lower_project(graph, false, channels)`, `sigmoid`. Drop `lower_offset_one` on this path.
3. Read: `Read` consumes the normalized stream `xn` times the gate and divides by `lanes` (mean). Extend the `Read` kernels' argument to carry the `1/lanes` scale, or add the scale as a scalar program; both backends must reduce in lane order as today.
4. Write gate: `lower_project(graph, false, lanes)` on the same `xn` node, scale by `1/lanes`, `sigmoid`, multiply by 2. `Outer` unchanged.
5. Head: `lower_collapse` uses steps 1 to 3 with no write gate.
6. `rank` 0 keeps its meaning: no Normalize node, gates fixed at one, `Read` is the mean of the raw lanes and `Outer` scales by one, so `hyper(1, 0, ...)` still equals `res(...)` bit for bit and a fresh model still starts at the plain residual.
7. Backward through RMS, silu, sigmoid and the mean scale on both backends; bundle text unchanged (`hyper,lanes,rank,blocks`).

### Acceptance (public Recipe CLI, real hardware)

- Regression: `recipe --device cpu` and `--device nv0` on archy, `hyper(1, 0, &m)` versus `res(...)` bit identical at 1, 5, 13 and 30 epochs on loss, all predictions, inference and every weight tensor, as in the #598 thread.
- Parameter accounting: `hyper(4, 320, &m)` on a 2560-wide stream reports exactly 10240 + 10240*320 + 320*10240 + 10240*4 mixer parameters per block and 10240 + 2*10240*320 at the head, matching the GGUF `hc_*` and `output_hc_*` element counts.
- Reference match: bind the hyper-connected GGUF checkpoint on archy (the one whose metadata declares 4 lanes and rank 320) to a one-layer `hyper(4, 320, &attn_or_delta)` program through the public entry, run one forward on a fixed token prefix on nv0 and on cpu, and compare `hc_mixed`, `hc_inject` and `hc_combine` at that layer against llama.cpp's `cb` dumps for the same prefix. Report the max absolute difference per tensor and the cpu/nv0 digests; require fp32-level agreement with llama.cpp and byte identity between cpu and nv0 at the determinism suite scale (131 rows).
- Lifecycle: train, save, resume, infer with `hyper(4, 320, ...)` and nonzero rank on archy nv0; all losses finite; resumed loss equals the saved final loss.
